requests/cookies.py

```diff
@@ -57,6 +57,12 @@ def get_full_url(self):
         # If they did set it, retrieve it and reconstruct the expected domain
         host = self._r.headers['Host']
         parsed = urlparse(self._r.url)
+
+        # If parsed url is str type, ensure that host is also str type
+        if isinstance(parsed.scheme, str) and not isinstance(host, str)\
+                and isinstance(host, bytes):
+            host = host.decode('ascii')
```

Comment:
Thanks for the tip. I missed the to_native_string utility function provided by requests.
I have made adjustments to this.

We may also want to consider making the import references in utils.py less explicit as to avoid issues with circular dependencies. utils.py imports some objects/functions from cookies explicitly with ```from .cookies import RequestsCookieJar, cookiejar_from_dict```. These might be better achieved as a more generic import of the cookies module ```from . import cookies```.

This would allow us to be more explicit in our imports within cookies.py, ```from .utils import to_native_string``` versus ```from . import utils```, which is perhaps more beneficial?